internal/core/services/worker/worker.go

```diff
@@ -0,0 +1,122 @@
+package worker
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/topfreegames/maestro/internal/core/entities/operation"
+	"github.com/topfreegames/maestro/internal/core/operations"
+	"github.com/topfreegames/maestro/internal/core/ports"
+	"github.com/topfreegames/maestro/internal/core/services/operation_manager"
+	"github.com/topfreegames/maestro/internal/core/services/operations_registry"
+	"go.uber.org/zap"
+)
+
+// Worker is the service responsible for implemeting the worker
+// responsabilities.
+type Worker struct {
+	opManager *operation_manager.OperationManager
+	// TODO(gabrielcorado): check if we this is the right place to have all
+	// executors.
+	executors map[string]operations.Executor
+
+	logger *zap.Logger
+}
+
+type WorkerOptions struct {
+	OperationStorage   ports.OperationStorage
+	OperationFlow      ports.OperationFlow
+	Executors          []operations.Executor
+	OperationsRegistry *operations_registry.Registry
+}
+
+func NewWorker(opts *WorkerOptions) *Worker {
+	opManager := operation_manager.New(opts.OperationFlow, opts.OperationStorage)
+	if opts.OperationsRegistry != nil {
+		opManager = operation_manager.NewWithRegistry(opts.OperationFlow, opts.OperationStorage, *opts.OperationsRegistry)
+	}
+
+	executors := make(map[string]operations.Executor)
+	for _, executor := range opts.Executors {
+		// TODO(gabrielcorado): are we going to receive the executor
+		// initialized?
+		executors[executor.Name()] = executor
+	}
+
+	return &Worker{
+		opManager: opManager,
+		executors: executors,
+		logger:    zap.L().With(zap.String("service", "worker")),
```

Comment:
Not 100%. I wanted to add something like the caller but the more high level to identify which component those logs are coming from. Also, I don't want to rely on the file names since we can split this component into more files or even packages.

Reply:
Got it, my concern is the need to set this label for every object/service/adapter that uses zap for logs. If it's not default, ok!

Reply:
It makes sense. We could start adding logs to multiple services to see if we can develop a solution for all of them.